ext/com/concurrent_ruby/ext/SynchronizationLibrary.java

```diff
@@ -55,14 +55,14 @@ public IRubyObject initialize(ThreadContext context, IRubyObject[] args, Block b
             }
         }
 
-        @JRubyMethod(name = "synchronize", visibility = Visibility.PRIVATE)
+        @JRubyMethod(name = "synchronize", visibility = Visibility.PROTECTED)
```

Comment:
Actually I am not sure about this code now. I was unable to get a module to dispatch to a private method even on a hierarchy that had included it. Further, I was unable to get a dotted form (foo.bar) to dispatch to a parent's private methods. Only the bare version worked:

```
[] ~/projects/jruby $ rvm ruby-2.2 do ruby -e "class A; private def bar; end; end; class B < A; def foo; bar; end; end; B.new.foo"

[] ~/projects/jruby $ rvm ruby-2.2 do ruby -e "class A; private def bar; end; end; class B < A; def foo; self.bar; end; end; B.new.foo"
-e:1:in `foo': private method `bar' called for #<B:0x007fc681927300> (NoMethodError)
	from -e:1:in `<main>'

[] ~/projects/jruby $ rvm ruby-2.2 do ruby -e "class A; private def bar; end; end; class B < A; def foo; (a = self; a).bar; end; end; B.new.foo"
-e:1:in `foo': private method `bar' called for #<B:0x007fe82a0df218> (NoMethodError)
	from -e:1:in `<main>'
```